### PR TITLE
Update C++/Obj C++ support for Firebase 9

### PR DESCRIPTION
**Pull request by paulb777**
- Fix ObjC++ Firebase.h regression
- Copy and adapt Storage and Functions ObjC API tests for Objective C++
- Several Firebase 9 release notes updates

### CODE REVIEW

**Comment by paulb777** [comment]
cc: @jonsimantov

File: FirebaseCore/CHANGELOG.md
Comment (morganchen12) [inline FirebaseCore/CHANGELOG.md:16]: I think you can just say `C++` here, since that should encompass both ObjC++ and C++.

File: FirebaseStorage.podspec
Comment (morganchen12) [inline FirebaseStorage.podspec:57]: nit: you can use `{m,mm}` to be more exact

**Review by morganchen12 — COMMENTED** [review]
Dev materials LGTM, please wait for other approvers before merging.

File: FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (ryanwilson) [inline FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm:123]: Should be removed I think?

File: FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (ryanwilson) [inline FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm:117]: CocoaPods only?

Comment (paulb777) [reply FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm:117]: Yes, there's no way to make public typedefs in SwiftPM

File: FirebaseCore/CHANGELOG.md
Comment (paulb777) [reply FirebaseCore/CHANGELOG.md]: Did both to avoid confusion.

File: FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (paulb777) [reply FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm]: Done in both the ObjC and ObjC++ tests

File: FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (ryanwilson) [reply FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm:117]: Ah right. I guess these tests won't run in SwiftPM though right?

**Review by ryanwilson — APPROVED** [review]

File: FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (paulb777) [reply FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm:117]: Right. The `#ifdef` enables the rest of the file to run in SwiftPM

Comment (ryanwilson) [reply FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm:117]: We're excluding the file entirely in SwiftPM, so I think it'd be fine to just eliminate the CocoaPods check but I'm fine leaving it for when we can eventually get support 

https://github.com/firebase/firebase-ios-sdk/blob/8623b798cbabaecedadb6321b6dad2c11d67d450/Package.swift#L1110-L1113

**Pull request merged**
